commit ffa6a2fa83452afe6a6ec67c4a3e8a5f1a6aaeef
Author: Jean-Baptiste Kempf <jb@videolan.org>
Date:   Mon Feb 15 23:50:27 2010 +0100

    Png: compatibility with libpng 1.4
    
    Patch probably from arekm -- pld-linux . org

Index: vlc-1.0.5/modules/codec/png.c
===================================================================
--- vlc-1.0.5.orig/modules/codec/png.c
+++ vlc-1.0.5/modules/codec/png.c
@@ -155,7 +155,7 @@ static picture_t *DecodeBlock( decoder_t
     p_info = png_create_info_struct( p_png );
     if( p_info == NULL )
     {
-        png_destroy_read_struct( &p_png, png_infopp_NULL, png_infopp_NULL );
+        png_destroy_read_struct( &p_png, NULL, NULL );
         block_Release( p_block ); *pp_block = NULL;
         return NULL;
     }
@@ -163,7 +163,7 @@ static picture_t *DecodeBlock( decoder_t
     p_end_info = png_create_info_struct( p_png );
     if( p_end_info == NULL )
     {
-        png_destroy_read_struct( &p_png, &p_info, png_infopp_NULL );
+        png_destroy_read_struct( &p_png, &p_info, NULL );
         block_Release( p_block ); *pp_block = NULL;
         return NULL;
     }
